ui/lib/pki/addon/components/page/pki-issuer-edit.ts

```diff
@@ -0,0 +1,57 @@
+import Component from '@glimmer/component';
+import { tracked } from '@glimmer/tracking';
+import { inject as service } from '@ember/service';
+import { action } from '@ember/object';
+import { task } from 'ember-concurrency';
+import { waitFor } from '@ember/test-waiters';
+import errorMessage from 'vault/utils/error-message';
+import RouterService from '@ember/routing/router-service';
+import FlashMessageService from 'vault/services/flash-messages';
+import PkiIssuerModel from 'vault/models/pki/issuer';
+
+interface Args {
+  model: PkiIssuerModel;
+}
+
+export default class PkiIssuerEditComponent extends Component<Args> {
+  @service declare readonly router: RouterService;
+  @service declare readonly flashMessages: FlashMessageService;
+
+  @tracked usageValues: Array<string> = [];
+  @tracked error = null;
+
+  constructor(owner: unknown, args: Args) {
+    super(owner, args);
+    this.usageValues = (this.args.model.usage || '').split(',');
+  }
+
+  toDetails() {
+    this.router.transitionTo('vault.cluster.secrets.backend.pki.issuers.issuer.details');
```

Comment:
you mixed it up here! usually I've seen you pass the transition action as an `@onSave` arg to the component. 💭

Reply:
Hehe. Ya I suppose I did :grin:. I guess in the last couple of projects where we have started using this page component pattern I've handled more things internally since the component is only used in the route template.